src/main.rs

```diff
@@ -464,10 +466,46 @@ enum TestOutcome {
 }
 
 impl Toolchain {
+    fn default_nightly() -> Option<Date<Utc>> {
+        if let Ok(output) = Command::new("rustc").arg("-vV").output() {
+            let text = &String::from_utf8_lossy(&output.stdout);
+            let regex = Regex::new(r"(?m)^release: .*-nightly$").unwrap();
+
+            if regex.is_match(text) {
+                let regex = Regex::new(r"(?m)^commit-date: (\d{4})-(\d{2})-(\d{2})$").unwrap();
+                if let Some(cap) = regex.captures(text) {
+                    let year = cap.get(1)?.as_str().parse::<i32>().ok()?;
+                    let month = cap.get(2)?.as_str().parse::<u32>().ok()?;
+                    let day = cap.get(3)?.as_str().parse::<u32>().ok()?;
+
+                    return Some(Date::from_utc(
+                        naive::NaiveDate::from_ymd(year, month, day),
+                        Utc,
+                    ));
+                }
+            }
+        }
```

Comment:
Let's maybe try to use `rustc-version` crate here? It seems to take care of the low-level details here for us